src/v/archival/segment_reupload.cc

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "segment_reupload.h"
+
+#include "cloud_storage/partition_manifest.h"
+#include "logger.h"
+#include "storage/disk_log_impl.h"
+#include "storage/fs_utils.h"
+
+namespace archival {
+segment_collector::segment_collector(
+  model::offset begin_inclusive,
+  const cloud_storage::partition_manifest* manifest,
```

Comment:
nit: great to avoid raw pointers whenever possible

Reply:
sure, fixed in current PR [here](https://github.com/redpanda-data/redpanda/pull/6455/commits/837a57efbbedaefaa52e430f01f479cfb1a51fb0)